app/src/main/java/com/internship/move/feature/map/MapFragment.kt

```diff
@@ -24,16 +24,12 @@ class MapFragment : Fragment(R.layout.fragment_map) {
             findNavController().navigate(MapFragmentDirections.actionMapFragmentToNavigationIntro())
         }
 
-        binding.ClearBtn.setOnClickListener {
-            initPersistence()
+        binding.clearBtn.setOnClickListener {
```

Comment:
Move the preferences out of the UI. 
In this case, the Preferences are Business Logic and the Fragment is UI, so they shouldn't mix together.
USe the ViewModel to access the SharedPrefs.